addons/im_livechat/controllers/main.py

```diff
@@ -95,7 +95,11 @@ def get_session(self, channel_id, anonymous_name, **kwargs):
             country = request.env['res.country'].sudo().search([('code', '=', country_code)], limit=1) if country_code else None
             if country:
                 anonymous_name, country_id = _("%s (%s)") % (anonymous_name, country.name), country.id
-        return request.env["im_livechat.channel"].with_context(lang=False).get_mail_channel(channel_id, anonymous_name, user_id, country_id)
+
+        if previous_operator_id:
+            previous_operator_id = int(previous_operator_id)
+
+        return request.env["im_livechat.channel"].with_context(lang=False).sudo().browse(channel_id)._get_mail_channel(anonymous_name, previous_operator_id, user_id, country_id)
```

Comment:
Just be sure this sudo as no side effect (notably: security, leak, multi company)

Reply:
fix with the proxy method on website that we discussed. Changes are included in the REF commit (1st one).